File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        counter_map = {}
        hasDuplicate = False
        for i in range(0, len(nums)):
            if counter_map.get(nums[i]) == None:
                counter_map[nums[i]] = 1;
            else:
                hasDuplicate = True
        return hasDuplicate

        

File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] nums, int target) {
        if(nums.length == 0 ) {
            return -1;
        }

        return find(nums, 0, nums.length - 1, target);
       
    }

    public int find(int[] nums, int beg, int end, int target) {
        int middle = (Math.abs(beg) + Math.abs(end)) / 2;   
        
        if(nums[middle] == target) {
            return middle;
        }

        if(end <= beg) {
            return -1;
        }
    
        if(nums[middle] < target) {
            return find(nums, middle + 1, end, target);
        } else {
            return find(nums, beg, middle - 1, target);
        }
    }   
}
